libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -714,24 +716,56 @@ func `/`(a, b: Duration): float64 =
     fb = float64(b.nanoseconds)
   fa / fb
 
+proc disconnectPeer(g: GossipSub, peer: PubSubPeer) {.async.} =
+  when defined(libp2p_agents_metrics):
+    let agent =
+      block:
+        if peer.shortAgent.len > 0:
+          peer.shortAgent
+        else:
+          if peer.sendConn != nil:
+            let shortAgent = peer.sendConn.peerInfo.agentVersion.split("/")[0].toLowerAscii()
+            if KnownLibP2PAgentsSeq.contains(shortAgent):
+              peer.shortAgent = shortAgent
+            else:
+              peer.shortAgent = "unknown"
+            peer.shortAgent
+          else:
+            "unknown"
+    libp2p_gossipsub_bad_score_disconnection.inc(labelValues = [agent])
+  else:
+    libp2p_gossipsub_bad_score_disconnection.inc(labelValues = ["unknown"])
+
+  if peer.sendConn != nil:
```

Comment:
Interesting, I imagine that this will be true for most peers? For example, we might have a send connection, but the peer might be faulty/bad or have flagged _us_ as such, in which case sending to it is pointless... So, maybe we shouldn't use this as a criterion?

Reply:
also, might be a good idea to cherry pick into its own PR.

Reply:
Just treating it as a unsafe pointer and being safe I guess :) you make a fair point tho